jobs/payment-jobs/tasks/ap_task.py

```diff
@@ -0,0 +1,183 @@
+            batch_total += rs.refund_amount
+            rs.status = RoutingSlipStatus.REFUND_UPLOADED.value
+        batch_trailer = cls.get_batch_trailer(batch_number, float(batch_total), control_total=total_line_count)
+        ap_content = f'{ap_content}{batch_trailer}'
+
+        cls._create_file_and_upload(ap_content)
+
+    @classmethod
+    def _create_non_gov_disbursement_file(cls):  # pylint:disable=too-many-locals
+        """Create AP file for disbursement for non government entities without a GL code via EFT and upload to CGI."""
+        cls.ap_type = EjvFileType.NON_GOV_DISBURSEMENT
+        bca_partner = CorpTypeModel.find_by_code('BCA')
+        total_invoices: List[InvoiceModel] = EjvPartnerDistributionTask().get_invoices_for_disbursement(bca_partner) + \
+            EjvPartnerDistributionTask().get_invoices_for_refund_reversal(bca_partner)
+
+        current_app.logger.info(f'Found {len(total_invoices)} to disburse.')
+        if not total_invoices:
+            return
+
+        # 250 MAX is all the transactions the feeder can take per batch.
+        for invoices in batched(total_invoices, 250):
+            bca_distribution = cls._get_bca_distribution_string()
+            ejv_file_model: EjvFileModel = EjvFileModel(
+                file_type=cls.ap_type.value,
+                file_ref=cls.get_file_name(),
+                disbursement_status_code=DisbursementStatus.UPLOADED.value
+            ).flush()
+            # Create a single header record for this file, provides query ejv_file -> ejv_header -> ejv_invoice_links
+            # Note the inbox file doesn't include ejv_header when submitting.
+            ejv_header_model: EjvFileModel = EjvHeaderModel(
+                partner_code='BCA',
+                disbursement_status_code=DisbursementStatus.UPLOADED.value,
+                ejv_file_id=ejv_file_model.id
+            ).flush()
+
+            batch_number: str = cls.get_batch_number(ejv_file_model.id)
+            ap_content: str = cls.get_batch_header(batch_number)
+            batch_total = 0
+            control_total: int = 0
+            for inv in invoices:
+                disbursement_invoice_total = inv.total - inv.service_fees
+                batch_total += disbursement_invoice_total
+                if disbursement_invoice_total == 0:
+                    continue
+                ap_content = f'{ap_content}{cls.get_ap_header(disbursement_invoice_total, inv.id, inv.created_on)}'
+                control_total += 1
+                line_number: int = 0
+                for line_item in inv.payment_line_items:
+                    if line_item.total == 0:
+                        continue
+                    ap_line = APLine.from_invoice_and_line_item(inv, line_item, line_number + 1, bca_distribution)
+                    ap_content = f'{ap_content}{cls.get_ap_invoice_line(ap_line)}'
+                    line_number += 1
+                control_total += line_number
```

Comment:
so control_total is 1 + the number of lineItems? Is that correct or should it be == number of lineItems?

Reply:
control_total gets incremented when an AP line is created, I'm just adding the number of payment_line_items to the control_total because we need a control line each time a payment line item is added.